data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/source/oteltrace/OTelTraceSourceConfig.java

```diff
@@ -61,18 +69,38 @@ private OTelTraceSourceConfig(final int requestTimeoutInMillis,
         this.sslKeyCertChainFile = sslKeyCertChainFile;
         this.sslKeyFile = sslKeyFile;
         this.acmCertificateArn = acmCertificateArn;
+        this.acmPrivateKeyPassword = acmPrivateKeyPassword;
+        this.awsRegion = awsRegion;
         this.threadCount = threadCount;
         this.maxConnectionCount = maxConnectionCount;
-        if (useAcmCertForSSL && (acmCertificateArn == null || acmCertificateArn.isEmpty())) {
-            throw new IllegalArgumentException(String.format("%s is enabled, %s can not be empty or null", USE_ACM_CERT_FOR_SSL, ACM_CERT_ARN));
-        } else {
-            if (ssl && (sslKeyCertChainFile == null || sslKeyCertChainFile.isEmpty())) {
-                throw new IllegalArgumentException(String.format("%s is enabled, %s can not be empty or null", SSL, SSL_KEY_CERT_FILE));
-            }
-            if (ssl && (sslKeyFile == null || sslKeyFile.isEmpty())) {
-                throw new IllegalArgumentException(String.format("%s is enabled, %s can not be empty or null", SSL, SSL_KEY_CERT_FILE));
+        boolean certAndKeyFileInS3 = false;
+        if (useAcmCertForSSL) {
+            validateSSLArgument(String.format("%s is enabled", USE_ACM_CERT_FOR_SSL), acmCertificateArn, ACM_CERT_ARN);
+            validateSSLArgument(String.format("%s is enabled", USE_ACM_CERT_FOR_SSL), awsRegion, AWS_REGION);
+        } else if(ssl) {
+            validateSSLCertificateFiles();
+            certAndKeyFileInS3 = isSSLCertificateLocatedInS3();
+            if (certAndKeyFileInS3) {
+                validateSSLArgument("The certificate and key files are located in S3", awsRegion, AWS_REGION);
             }
         }
+        this.sslCertAndKeyFileInS3 = certAndKeyFileInS3;
+    }
+
+    private void validateSSLArgument(final String sslTypeMessage, final String argument, final String argumentName) {
+        if (argument == null || argument.isEmpty()) {
```

Comment:
nit - can use `StringUtils.isEmpty()` (have started to use the micrometer impl elsewhere in the project.

Reply:
Sure, looks like we already have org.apache.commons:commons-lang3:3.11. I add this to dependency and we can use ```StringUtils.isEmpty()```